### PR TITLE
Swift and Go examples for fee calc and lsp connection

### PR DESCRIPTION
**Pull request by ubbabeck**

### CODE REVIEW

**Comment by ubbabeck** [comment]
I have ran all the examples for python and swift, but not go. The reason for this is that I had some issues with my environment so the code did not run for some reason.

**Review by roeierez — APPROVED** [review]
LGTM

File: src/guide/receive_onchain.md [hunk 1]
```diff
@@ -384,4 +448,65 @@ def calculate_fees_for_amount(amount_sats):
...
+
+When the amount to be received exceeds the inbound liquidity of the node, a new channel will be opened by the LSP in order for the node to receive it. This can checked by retrieving the NodeState from the SDK and comparing the inbound liquidity to the amount to be received. If the amount is greater or equal to the inbound liquidity, a new channel opening is required.
+
+To calculate the fees for a channel being opened by the LSP:
+
+```go
+func CalculateChannelOpeningFee(amountSats uint64) (uint64, error) {
+	isChannelOpeningFeeNeeded := isChannelOpeningFeeNeeded(amountSats)
+	if !isChannelOpeningFeeNeeded {
+		return 0, fmt.Error("Channel not needed")
```
Comment (dangeross) [inline src/guide/receive_onchain.md:462]: Should be `Errorf`

**Review by dangeross — CHANGES_REQUESTED** [review]
LGTM, just needs one small change

File: src/guide/receive_onchain.md [hunk 2]
```diff
@@ -318,15 +382,15 @@ This information can be retrieved for each LSP and then calculated:
...
     LSPInformation? lspInformation = await fetchLspInfo(lspId!);
 
-    // setupFee is the proportional fee charged based on the amount
+    // setupFee is the proportional fee charged based on the amount.
     int setupFee = (lspInformation.channelFeePermyriad / 100);
-    // minFee is the minimum fee required by the LSP to open a channeş
+    // minFee is the minimum fee required by the LSP to open a channel.
     int minFee = lspInfo.channelMinimumFeeMsat ~/ 1000;
-    // A setup fee of {setupFee}% with a minimum of {minFee} will be applied for sending more than {liquidity}.
+    // A setup fee of {setupFee} with a minimum of {minFee} will be applied for sending more than {liquidity}.
```
Comment (erdemyerebasmaz) [inline src/guide/receive_onchain.md:393]: The percentage is not a typo, setup fee is the proportional fee, e.g. 7.5%

File: src/guide/receive_onchain.md (hunk 1 above)
Comment (ubbabeck) [reply src/guide/receive_onchain.md]: fixed

**Review by dangeross — APPROVED** [review]

**Pull request merged**